src/core/network.js

```diff
@@ -454,7 +714,7 @@ var NetworkManager = (function NetworkManagerClosure() {
         networkManager.abortRequest(fullRequestXhrId);
       }
 
-      this._headersReceivedCapability.resolve();
+      this._headersReceivedCapability.resolve(headers);
```

Comment:
Don't leak the headers here. Any desired information should have been parsed by now and exposed as a property of `this`, eg `this.filename`